main/src/main.rs

```diff
@@ -71,31 +73,50 @@ fn main() -> eyre::Result<()> {
     check_writable_dir(&config.state_dir)
         .wrap_err("state-dir value in a config should be a writeable directory path")?;
 
-    let tokio_cores = config
-        .rpc_endpoint
-        .utility_cores_ids
-        .iter()
-        .cloned()
-        .map(Into::into)
-        .collect::<Vec<_>>();
+    let tokio_cores = config.rpc_endpoint.utility_cores_ids.clone();
+
+    let tokio_core_ids_state = CpuIdsHandle::new(tokio_cores);
+    let tokio_core_ids_state2 = tokio_core_ids_state.clone();
+    let tokio_core_ids_state3 = tokio_core_ids_state.clone();
 
     let runtime = tokio::runtime::Builder::new_multi_thread()
         .enable_all()
         .worker_threads(2)
         .on_thread_start(move || {
             let pid = std::thread::current().id();
+            let tokio_cores = tokio_core_ids_state.get();
             tracing::info!("Pinning tokio thread {pid:?} to cores {tokio_cores:?}");
-            if !cpu_utils::pinning::pin_current_thread_to_cpuset(tokio_cores.iter().copied()) {
+            if !cpu_utils::pinning::pin_current_thread_to_cpuset(tokio_cores.into_iter()) {
                 tracing::error!("Tokio thread pinning failed");
             }
         })
+        .on_thread_unpark(move || {
+            thread_local! {
+                static LAST_EPOCH: Cell<u32> = 0.into();
```

Comment:
let's not use epoch here, b/c it intersects with other meaning of epochs. Maybe smth like `LAST_SEEN_REPIN_COUNTER`.